File: src/fe/fakebook-aio/src/app/services/models/pipeline.model.ts
export interface PipelineCreateModel {
    jobName: string;
    jobDescription: string;
    authToken: string;
    pipelineContent: string;
}

export interface PipelineUpdateModel {
    jobName: string;
    jobDescription: string;
    authToken: string;
    pipelineContent: string;
}

export interface PipelineModel {
    id: string,
    createdDate: Date,
    lastModifiedDate: Date,
    jobName: string,
    jobDescription: string,
    authToken: string,
    pipelineContent: string,
}

File: src/be/Services/Fakebook.AIO/Models/PipelineModel.cs
using Fakebook.AIO.Entity;

namespace Fakebook.AIO.Models;

public class PipelineCreateModel
{
    public string JobName { get; set; } = null!;
    public string JobDescription { get; set; } = null!;
    public string AuthToken { get; set; } = null!;
    public string PipelineContent { get; set; } = null!;

    public Pipeline ToPipeline()
    {
        return new Pipeline()
        {
            JobName = JobName,
            JobDescription = JobDescription,
            AuthToken = AuthToken,
            PipelineContent = PipelineContent,
        };
    }
}
public class PipelineUpdateModel
{
    public string JobName { get; set; } = null!;
    public string JobDescription { get; set; } = null!;
    public string AuthToken { get; set; } = null!;
    public string PipelineContent { get; set; } = null!;

    public Pipeline ToPipeline()
    {
        return new Pipeline()
        {
            JobName = JobName,
            JobDescription = JobDescription,
            AuthToken = AuthToken,
            PipelineContent = PipelineContent,
        };
    }
}